src/syscheckd/fim_db.c

```diff
@@ -1078,6 +1082,24 @@ void fim_db_remove_path(fdb_t *fim_sql, fim_entry *entry, pthread_mutex_t *mutex
         w_mutex_unlock(mutex);
 }
 
+void fim_db_process_path(fdb_t *fim_sql, fim_entry *entry, pthread_mutex_t *mutex, __attribute__((unused))void *arg) {
+
+    fim_event_mode *mode = (fim_event_mode *) arg;
+    int conf_file = fim_configuration_directory(entry->path, "file");
+
+    if(mode != FIM_REALTIME && syscheck.opts[conf_file] & REALTIME_ACTIVE) {
```

Comment:
This part of the code should be programmed using a `switch` statement. It's more efficient and makes more sense.